### ISSUE TITLE
Rule collision when multiple vendored projects contain executables with matching names

### ISSUE THREAD

**Issue by gridbugs** [ocaml/dune#6830]
## Expected Behavior

It should be possible to vendor multiple packages which contain executables whose names (`public_name` in dune files) are the same. The source code for a package may contain executables for testing and experimentation, but typically one vendors a package just to get access to the libraries it contains.

## Actual Behavior

When resolving install stanzas that target the bin section, or expanding `%{bin:...}` expressions, dune gives an error `Multiple rules generated ...`

## Reproduction

I've made a small project to reproduce the issue: https://github.com/gridbugs/dune-vendored-projects-rule-collision-repro

It vendors two packages: OSCADml.0.1.1  and scad_ml.1.1.0. Both those packages contain executables named "circle". The `dune` file contains an install stanza which installs a shell script into the bin section. Running `dune build @install` gives the error:
```
Error: Multiple rules generated for _build/install/default/bin/circle:
- vendored/scad_ml.1.1.0/test/dune:3
- vendored/OSCADml.0.1.1/test/dune:3
-> required by
   _build/default/dune-vendored-projects-rule-collision-repro.install
-> required by alias install
```

## Specifications

- Version of `dune` (output of `dune --version`): 3.6.1
- Version of `ocaml` (output of `ocamlc --version`): 5.0.0
- Operating system (distribution and version): NixOS (linux 6.1.1)

**Comment by rgrinberg** [comment]
How would we resolve `%{bin:circle}` given this setup?

**Comment by gridbugs** [comment]
If the `circle` target is introduced in a vendored project, maybe we should require it to be referred to with `%{bin:foo.circle}` in the main project.

**Comment by gridbugs** [comment]
On second thought I don't think namespacing executables like `foo.circle` will work. We want to be able to vendor build tools and refer to them in dune files as if they were installed globally. Instead would it make sense to give a way to tell dune to ignore or rename certain executables in vendored dirs?

**Comment by Alizter** [comment]
What if we have a way of masking executable names using the vendored_dirs stanza? This would also be useful for vendoring multiple versions of an executable together.

**Comment by rgrinberg** [comment]
> On second thought I don't think namespacing executables like foo.circle will work. We want to be able to vendor build tools and refer to them in dune files as if they were installed globally. Instead would it make sense to give a way to tell dune to ignore or rename certain executables in vendored dirs?

Yeah, that's probably the way to go. Although currently we are also setting up install rules for these vendored build tools and that's definitely wrong and probably the first thing to fix.
